### PR TITLE
Readd module _NIOConcurrency to not break adopters

### PR DESCRIPTION
**Pull request by fabianfett**

### CODE REVIEW

**Review by Lukasa — APPROVED** [review]
LGTM. Let's not forget to reach out to our adopters.

File: Package.swift [hunk 1]
```diff
@@ -74,7 +76,7 @@ var targets: [PackageDescription.Target] = [
     .target(name: "NIOCrashTester",
             dependencies: ["NIOPosix", "NIOCore", "NIOEmbedded", "NIOHTTP1", "NIOWebSocket", "NIOFoundationCompat"]),
     .target(name: "NIOAsyncAwaitDemo",
-            dependencies: ["NIOPosix", "NIOCore", "NIOHTTP1"]),
+            dependencies: ["NIOPosix", "NIOCore", "NIOHTTP1", "_NIOConcurrency"]),
```
Comment (PeterAdams-A) [inline Package.swift:79]: Did you need to read this line?

**Comment by fabianfett** [comment]
@PeterAdams-A great catch. Fixed.

**Review by dnadoba — APPROVED** [review]

**Comment by fabianfett** [comment]
`swift:nightly-main` still falling because of `Sendable` requirements.

**Pull request merged**